test/lifecycle_test.dart

```diff
@@ -673,6 +676,46 @@ void sharedLifecycleTests<T extends react.Component>({
           ].where((matcher) => matcher != null).toList()));
     });
 
+    if (isComponent2) {
+      test('init testing', () {
+        const Map initialState = const {
+          'initialState': 'initial',
+        };
+
+        final Map initialProps =
+            unmodifiableMap({'init': (react.Component2 component) => component.state = initialState});
+        final Map expectedProps = unmodifiableMap(defaultProps, initialProps, emptyChildrenProps);
+        LifecycleTestHelper component = getDartComponent(render(LifecycleTest(initialProps)));
+
+        expect(
+            component.lifecycleCalls,
+            equals([
+              matchCall('init', props: expectedProps),
```

Comment:
Since we're dealing with state mutation, these should all probably have `state: ` expectations as well

Reply:
It seems that getInitialState gets in the way is resetting the state to be an empty map, is this ok?

Reply:
No, we don't want that to happen, since we want consumers that don't override `getInitialState` to be able to initialize state in `init`.